wagtail/wagtailusers/forms.py

```diff
@@ -184,8 +184,9 @@ def clean_password2(self):
     def save(self, commit=True):
         user = super(UserEditForm, self).save(commit=False)
 
-        # users can access django-admin iff they are a superuser
-        user.is_staff = user.is_superuser
+        # users can access django-admin if they are a superuser
+        if not hasattr(user.__class__, 'is_staff') and 'is_staff' in vars(user):
```

Comment:
Surely this is a classic case to apply the EAFP principle? Try to set the property, but catch the `AttributeError` (or whatever the exception may be)...